packages/winston-log-enricher/lib/createFormatter.js

```diff
@@ -51,6 +51,12 @@ module.exports = function createFormatter(newrelic) {
       info[m] = metadata[m]
     })
 
+    // if the agent has a log aggregator,
+    // add log lines to it.
+    if (newrelic.agent.logs) {
+      newrelic.agent.logs.add(info)
```

Comment:
I see this is still in draft but wanted to call out the signature mismatch.  the signature of the log aggregator requires transaction context to get priority.  also the spec states we only want to add logs that are in transaction context.  When I wired this up to do perf testing I added it like this

```js
 58 +    const segment = newrelic.shim.getActiveSegment()
 59 +    if (newrelic.agent.logs && segment && segment.transaction) {
 60 +      newrelic.agent.logs.add(info, segment.transaction.priority)
 61 +    }
```

Reply:
Ah, great. Thanks for the info.

> also the spec states

I was working off the linked issue. Can you link me this spec?

Reply:
Behind the VPN, it's here: https://source.datanerd.us/agents/agent-specs/pull/570/files#diff-22ea975b3218ea7ba317770796151db7a25e9e8a59ef0799d8223ee8506802c9R50

Reply:
also a reminder you'll also need to check for the `config.application_logging && config.application.logging.enabled && config.application_logging.forwarding && config.application_logging.forwarding.enabled`